src/azul/debug.py

```diff
@@ -0,0 +1,66 @@
+from contextlib import (
+    ContextDecorator,
+)
+from logging import (
+    getLogger,
+)
+
+from azul import (
+    config,
+)
+
+log = getLogger(__name__)
+
+
+class RemoteDebugSession(ContextDecorator):
+    """
+    Adapted from https://stackoverflow.com/a/60333250/7830612
+
+    We use a context manager since the "debug session is over long-running tcp
+    connection. It needs to be closed explicitly in your lambda otherwise the
+    lambda will timeout."
+
+    For usage instructions, see the README.
+    """
+
+    def __init__(self):
+        self.active = False
+        if not self.is_available():
+            self.pydevd = None
+            log.warning('Remote debugging is not available')
+        else:
+            import pydevd
+            self.pydevd = pydevd
+
+    def __enter__(self):
+        if self.is_available():
+            self.pydevd.settrace(config.debug_host,
+                                 port=config.debug_port,
+                                 suspend=False,
+                                 stdoutToServer=True,
+                                 stderrToServer=True)
+            log.info('Starting remote debugging session')
+            self.active = True
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if self.active:
+            log.info(f'Stopping remote debugging on {config.debug_host}:{config.debug_port}')
+            self.pydevd.stoptrace()
+            self.active = False
+        return False
+
+    @classmethod
+    def is_available(cls):
+        try:
+            import pydevd  # noqa: F401
+        except ImportError:
+            available = False
+        else:
+            try:
+                _ = config.debug_host
```

Comment:
I don't know what the preferred way to write this check is, but the line `config.debug_host()` or `getattr(config, 'debug_port')` will also trigger the `KeyError` and perhaps looks less surprising, in case you hadn't considered it.

Reply:
I did `config.debug_host()` originally, but this triggered a PyCharm warning for "Statement doesn't do anything" or something like that. I'll try:

```
# noinspection PyStatementEffect
config.debug_host
# noinspection PyStatementEffect
config.debug_port
```
and we can see what @hannes-ucsc has to say about it.

Reply:
I'm increasingly confident you actually got it right the first time, and that my suggestions were a mistake. I think that this:

https://github.com/DataBiosphere/azul/blob/60203d179722d4728dd1326ec27a59962c807b9c/scripts/dss_v2_apdapter.py#L147 

means that _-assignment approach has precedent. I'm approving this PR since I might be wrong, but I encourage you to consider reverting to your initial approach.